packages/mattermost-redux/src/reducers/entities/users.ts

```diff
@@ -419,6 +419,17 @@ function profilesInGroup(state: RelationOneToMany<Group, UserProfile> = {}, acti
     }
 }
 
+function currentUserStatus(state: RelationOneToOne<UserProfile, string> = {}, action: GenericAction) {
```

Comment:
Why store the current user's status separately from the other users' statuses?

Reply:
Because the store structure is different. For other users we only store the string representation (`online`, `offline`, 'away` or `dnd`) whereas for the current user we need the full status object which contains custom status details as well.

Reply:
Ah right. I forgot that we still only store the status string for other users. There's designs around for using some of the other details in the profile popover, but I guess we didn't do that.